apps/test/unit/StudioAppTest.js

```diff
@@ -390,6 +391,167 @@ describe('StudioApp', () => {
     });
   });
 
+  describe('getCode', () => {
+    it('should get the starting blocks if the source is hidden', () => {
+      studioApp().editCode = true;
+      studioApp().hideSource = true;
+      studioApp().startBlocks_ = 'start blocks';
```

Comment:
Already figure this out! I do call `stubStudioApp` but what you need to do is actually use the studioAppSingleton instead of instantiating if you are messing about with properties apparently. https://github.com/code-dot-org/code-dot-org/blob/ed5f33b869cadd0a5082964c6319e924f68bd000/apps/test/unit/templates/CodeWorkspaceTest.js#L32-L40 does this.

Reply:
My new tests actually caused these unrelated `CodeWorkspace` tests to fail and adding the stubs to these tests allowed them to pass.